src/core/reactor_backend.cc

```diff
@@ -1167,9 +1167,10 @@ reactor_backend_selector reactor_backend_selector::default_backend() {
 
 std::vector<reactor_backend_selector> reactor_backend_selector::available() {
     std::vector<reactor_backend_selector> ret;
-    if (detect_aio_poll() && has_enough_aio_nr()) {
+    if (has_enough_aio_nr() && detect_aio_poll()) {
         ret.push_back(reactor_backend_selector("linux-aio"));
     }
+    seastar_logger.warn("epoll backend selected, this usually means the running system lacks AIO contexts. Consider increasing the value of fs.aio-max-nr.");
```

Comment:
The message is misleading, this function doesn't select the backend, but just reports back the available options.

Reply:
You're right, fixed.